src/aliceVision/keyframe/KeyframeSelector.cpp

```diff
@@ -30,448 +32,397 @@ namespace keyframe {
  */
 int getRandomInt()
 {
-  std::random_device rd;  // will be used to obtain a seed for the random number engine
-  std::mt19937 randomTwEngine(rd()); // standard mersenne_twister_engine seeded with rd()
-  std::uniform_int_distribution<> randomDist(0, std::numeric_limits<int>::max());
-  return randomDist(randomTwEngine);
+    std::random_device rd;             // will be used to obtain a seed for the random number engine
+    std::mt19937 randomTwEngine(rd()); // standard mersenne_twister_engine seeded with rd()
+    std::uniform_int_distribution<> randomDist(0, std::numeric_limits<int>::max());
+    return randomDist(randomTwEngine);
 }
 
-KeyframeSelector::KeyframeSelector(const std::vector<std::string>& mediaPaths,
-                                   const std::string& sensorDbPath,
-                                   const std::string& voctreeFilePath,
-                                   const std::string& outputFolder)
-  : _mediaPaths(mediaPaths)
-  , _sensorDbPath(sensorDbPath)
-  , _voctreeFilePath(voctreeFilePath)
-  , _outputFolder(outputFolder)
+double computeSharpness(const cv::Mat& grayscaleImage, const int window_size)
 {
-  if((_maxOutFrame != 0) &&
-     !_hasSharpnessSelection &&
-     !_hasSparseDistanceSelection)
-  {
-      ALICEVISION_LOG_ERROR("KeyframeSelector needs at least one selection method if output frame limited !");
-      throw std::invalid_argument("KeyframeSelector needs at least one selection method if output frame limited !");
-  }
-
-  // load vocabulary tree
-  _voctree.reset(new aliceVision::voctree::VocabularyTree<DescriptorFloat>(voctreeFilePath));
-
-  {
-      ALICEVISION_LOG_INFO("vocabulary tree loaded with :" << std::endl
-                       << "\t- " << _voctree->levels() << " levels" << std::endl
-                       << "\t- " << _voctree->splits() << " branching factor" << std::endl);
-  }
-
-  // check number of input media filePaths
-  if(mediaPaths.empty())
-  {
-    ALICEVISION_LOG_ERROR("Cannot create KeyframeSelector without a media file path !");
-    throw std::invalid_argument("Cannot create KeyframeSelector without a media file path !");
-  }
-
-  // resize mediasInfo container
-  _mediasInfo.resize(mediaPaths.size());
-
-  // create SIFT image describer
-  _imageDescriber.reset(new feature::ImageDescriber_SIFT());
-}
+    cv::Mat sum, sumsq, laplacian;
+    cv::Laplacian(grayscaleImage, laplacian, CV_64F);
+    cv::integral(laplacian, sum, sumsq);
 
-void KeyframeSelector::process()
-{
-  // create feeds and count minimum number of frames
-  std::size_t nbFrames = std::numeric_limits<std::size_t>::max();
-  for(std::size_t mediaIndex = 0; mediaIndex < _mediaPaths.size(); ++mediaIndex)
-  {
-    const auto& path = _mediaPaths.at(mediaIndex);
+    double n = window_size * window_size;
+    double maxstd = 0.0;
+    for (int i = 0; i < sum.rows - window_size; i++)
```

Comment:
It would be better for readability if the current row and column numbers were called `y` and `x` not `i` and `j`.

Reply:
Done.